### ISSUE TITLE
Programs crash with core dumps

### ISSUE THREAD

**Issue by MARTIMM** [MARTIMM/perl6-gnome-gtk3#9]
Crashes take place with some tryout programs of mine. It has several widgets from buttons to treeviews, so rather complex already.

The error is sometimes like
```
***MEMORY-ERROR***: qa-manager.pl6[10858]: GSlice: assertion failed: sinfo->n_allocated > 0
Abort (core dumped)
```
This is a nasty error for which I do not yet have an answer, let alone a solution :-(

**Comment by MARTIMM** [comment]
Most of the time it happens after closing a dialog or window

**Comment by MARTIMM** [comment]
Using the following code at the start of a program
```
use Gnome::N::X;
Gnome::N::debug(:on);
```
and setting an environment variable before running
```
setenv G_SLICE debug-blocks
```
pointed consistently to a specific location where a problem occurs but without the environment setting, it would be ignored until something bad happens.

**Comment by MARTIMM** [comment]
Looks like it is solved. I've used `.clear-object()` on iterators. It seems that those are owned by a **TreeModel**. The one time that an iterator can be cleaned up is when it is copied. The other problem was caused by creating a path from a native object which was handed down to a callback handler and clean up after use. That was a mistake because it was owned by the caller.

Most importantly, it wasn't a bug in the library.
